core/thread.c

```diff
@@ -372,10 +396,13 @@ static const char *state_names[STATUS_NUMOF] = {
 
 const char *thread_state_to_string(thread_status_t state)
 {
-    const char *name = state_names[state] ? state_names[state] : NULL;
+    const char *name =  NULL;
+    if (state < STATUS_NUMOF){
+        name = state_names[state];
+    }
 
-    assert(name != NULL); /* if compiling with assertions, this is an error that
-                             indicates that the table above is incomplete */
+    /* if compiling with assertions, this is an error that indicates that the table above is incomplete */
```

Comment:
```suggestion
    /* if compiling with assertions, this is an error
     * that indicates that the table above is incomplete */
```

```suggestion
    /* compiling with assertions, this will fail if the table above is incomplete */
```